proposals/evaluation/uc-3-inheritance.md

```diff
@@ -0,0 +1,225 @@
+# Access Inheritance Use Cases
+
+This is part 3 of the [implementation specific use cases comparison](./use-cases.md).
+
+See also: [UCR 2.3 Collection resource inherited access](https://solid.github.io/authorization-panel/authorization-ucr/#uc-inheritance)
+
+
+## Setup
+
+The Weekly status collection is an `ldp:BasicContainer`, which contains other `ldp:BasicContainers`, one for each weekly meeting, into which further documents can be placed. 
+
+```turtle
+# Resource: </weekly-status/>
+<.>
+  a ldp:BasicContainer ;
+  ldp:contains <2021-04-28/>, <2021-05-05/>, <2021-05-12/> .
+```
+
+We have the following hierarchy of resources (shown in more detail in the UCR):
+
+```
+</weekly-status/>
+</weekly-status/2021-04-28/>
+</weekly-status/2021-04-28/report.md>
+</weekly-status/2021-05-05/>
+</weekly-status/2021-05-05/report.md>
+</weekly-status/2021-05-05/diagram.jpg>
+</weekly-status/2021-05-12/>
+```
+
+We omit here the access-control resources as those are proposal specific.
+
+## 1. Read access to a group on a collection of resources
+
+We want to enable read access to all resources contained in `</weekly-status/>` for a group of people (`ex:Alice` & `ex:Bob`).
+
+### ACP
+
+Bob and Alice are part of the agent matcher `</acp/research#m1>`:
+
+```turtle
+# Resource: </acp/research>
+<#m1>
+  a acp:AgentMatcher ;
+  acp:agent ex:Bob, ex:Alice .
+```
+
+The research policy `</acp/research#p1>` gives read access to all agents matched by `</acp/research#m1>`:
+
+```turtle
+# Resource: </acp/research>
+<#p1>
+  a acp:Policy ;
+  acp:anyOf <#m1> ;
+  acp:allow acl:Read .
+```
+
+The access control `</weekly-status/.acp#ac1>` applies to all resources contained in `</weekly-status/>` and enables read access to all agents matched by `</acp/research#m1>`:
+
+```turtle
+# Resource: </weekly-status/.acp>
+<#ac1>
+  a acp:AccessControl ;
+  acp:apply </acp/research#p1> ; # applies the policy to the container resource </weekly-status/>
+  acp:applyMembers </acp/research#p1> . # applies the policy to all resources contained by </weekly-status/>
+```
+
+An unauthenticated client making a GET on `</weekly-status/>` container will receive a `Link: <.acp>; rel="acl"` header in the `401` response that points to the above `<.acp>`. 
+This relation is what makes the contents of `</weekly-status/.acp>` authoritative, and is therefore the information the client can use to decide what credentials to present.
```

Comment:
My goal was to focus on exactly what is required, but I guess it makes sense to include some implicit requirements and use the functional requirements for describing atoms of functionality.

Reply:
I don't see why public read would be needed for this UC. I only suggest that if for some reason it would be needed, the policies should have access set accordingly. IMO we should just remove this sentences unless they are really relevant for this use case.

Reply:
@elf-pavlik I tried pencilling in some statements about access to the `</weekly-status/.acp>` resource and it makes everything more complex and confusing in my opinion.

A few clarifications:
1. This Use Case is not about controlling access to the access control resource.
2. The Access Control Resource would not be readable by anyone except for the owner if it has no relevant access control statements.
3. Your first comment applies to the WAC example too, where we're not portraying who has `acl:Control` access either.

Now, I hear you that effective Access Control Resource discovery is beyond this Use Case.

My plan is to merge the UCs as is. I actually have a draft UC0 "Effective access control resource discovery" that I want to submit specifically to address this problem. I'd like to extract this bit of the text and its relevant parts in a different PR.